src/common/XAuth.cpp

```diff
@@ -149,18 +154,28 @@ bool XAuth::writeCookieToFile(const QString &display, const QString &fileName,
     auth.data = cookie.data();
     auth.data_length = cookie.size();
 
+    errno = 0;
     if (XauWriteAuth(authFp, &auth) == 0) {
+        qWarning().nospace() << "XAuth::writeCookieToFile: XauWriteAuth(FamilyLocal) failed with errno=" << errno;
         return false;
     }
 
     // Write the same entry again, just with FamilyWild
     auth.family = FamilyWild;
     auth.address_length = 0;
+    errno = 0;
     if (XauWriteAuth(authFp, &auth) == 0) {
+        qWarning().nospace() << "XAuth::writeCookieToFile: XauWriteAuth(FamilyWild) failed with errno=" << errno;
+        return false;
+    }
+
+    auto result = fflush(authFp);
```

Comment:
Can be just `if (fflush(authFp) != 0)

Reply:
Yeah true. At first I was thinking maybe should log `fflush`  return value aswell and that's why I put it in separate variable but then looking at man-page it didn't look like it would contain anything useful so doesn't seem worth bothering with that.